src/components/CompareResults/ResultsTitle.tsx

```diff
@@ -0,0 +1,87 @@
+import { useState } from 'react';
+
+import EditIcon from '@mui/icons-material/EditOutlined';
+import { Input } from '@mui/material';
+import { style } from 'typestyle';
+
+import useRawSearchParams from '../../hooks/useRawSearchParams';
+import { FontsRaw, Colors } from '../../styles';
+
+export const ResultsTitle = () => {
+  const [searchParams, updateSearchParams] = useRawSearchParams();
+  const [resultsTitle, setResultsTitle] = useState(
+    searchParams.get('title') || 'Results',
+  );
+  const [isEditing, setIsEditing] = useState(false);
+
+  const styles = {
+    title: style({
+      ...FontsRaw.HeadingXS,
+      fontWeight: 700,
+      letterSpacing: '-0.01em',
+      margin: 0,
+    }),
+    editIcon: style({
+      marginLeft: '10px',
+      cursor: 'pointer',
+      color: Colors.LinkText,
+    }),
+    hide: style({
+      display: 'none',
+    }),
+  };
+
+  const handleResultsChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newTitle = event.target.value.trim();
+    setResultsTitle(newTitle);
+  };
+
+  const handleEdit = () => setIsEditing(true);
+
+  const handleEditComplete = (
+    event: React.FocusEvent | React.KeyboardEvent,
+  ) => {
+    if (
+      (event as React.KeyboardEvent).key === 'Enter' ||
```

Comment:
It would be good to handle 'Escape' as a 'cancel' operation as well. Then it would just setIsEditing but do not update anything otherwise.